### PR TITLE
Enhanced py-utils test_framework to generate test_report after tests execution.

### PR DESCRIPTION
**Pull request by veerendra-simha-garikipati**
Signed-off-by: Veerendra Garikipati <veerendra.garikipati@seagate.com>

# Problem Statement
- Enhance py-utils test_framework to generate test_report after tests execution.

# Design
-  https://jts.seagate.com/browse/EOS-23188

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: N
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: N
- [x] Confirm Testing was performed with installed RPM [Y/N]: Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: N
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]: 
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by veerendra-simha-garikipati** [comment]
Sample test_report.html:
image

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/test_framework/generate_test_report.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    Generates HTML test_report under /tmp.
+    """
+    result_table = ''
+    overall_status = 'Passed'
+    total_ts = 0
+    total_failed = 0
+    time_taken=0
+    for ts, value in test_result.items():
+        status = list(value.keys())[0]
+        duration = int(list(value.values())[0])
```
Comment (rahul-tripathi-git) [inline py-utils/src/test_framework/generate_test_report.py:78]: Is casting to list required? looks like an overhead. Please check.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]
Please use your best judgement to cater this comment.

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Pull request merged**